pymc3/model.py

```diff
@@ -687,16 +687,25 @@ def flatten(self, vars=None):
         ----------
         vars : list of variables or None
             if None, then all model.free_RVs are used for flattening input
+        order : ArrayOrdering
+            Optional, use predefined ordering
+        inputvar : tt.vector
+            Optional, use predefined inputvar
 
         Returns
         -------
         flat_view
         """
         if vars is None:
             vars = self.free_RVs
-        order = ArrayOrdering(vars)
-        inputvar = tt.vector('flat_view', dtype=theano.config.floatX)
-        inputvar.tag.test_value = flatten_list(vars).tag.test_value
+        if order is None:
+            order = ArrayOrdering(vars)
+        if inputvar is None:
+            inputvar = tt.vector('flat_view', dtype=theano.config.floatX)
+            if vars:
```

Comment:
it should not, I use if there for the case of empty vars. or else I get exception from flatten